src/dir.cr

```diff
@@ -255,29 +255,147 @@ class Dir
   end
 
   def self.glob(patterns : Enumerable(String))
-    paths = LibC::Glob.new
-    flags = LibC::GlobFlags::BRACE | LibC::GlobFlags::TILDE
-    errfunc = -> (_path : UInt8*, _errno : Int32) { 0 }
+    path = self.working_directory
+    regexes = patterns.map { |pattern| glob2regex(pattern) }
+    if patterns.any? { |ptrn| ptrn.includes? "**" }
+      recursion_depth = Int32::MAX
+    else
+      recursion_depth = patterns.map { |ptrn| ptrn.count(File::SEPARATOR) }.max
+    end
+    scandir(path, "", regexes, 0, recursion_depth) do |path|
+      yield path
+    end
+  end
+  
+  private def self.glob2regex(pattern)
+    if pattern.size == 0 || pattern == File::SEPARATOR
+      raise ArgumentError.new "Empty glob pattern"
+    end
+
+    # characters which are escapable by a backslash in a glob pattern;
+    # Windows paths must have double backslashes:
+    escapable = ['?', '{', '}', '*', ',', '\\']
+    # characters which must be escaped in a PCRE regex:
+    escaped = ['$', '(', ')', '+', '.', '[', '^', '|', '/']
```

Comment:
Basically because they're immutable, so you don't accidentally reassign or something.